### DISCUSSION TITLE
Why `watson()` method needed?

### DISCUSSION THREAD

**Discussion by itlancer** [airsdk/Adobe-Runtime-Support#3344] [Q&A]
Why `watson()` method needed?
https://airsdk.dev/reference/actionscript/3.0/package-detail.html
It not documented in AIR reference but IDE (VSCode) offer such hint:
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/10899066/031451a6-a9eb-4450-9e2d-eafbe961aa96)
If it legacy and do nothing may be it should be deleted from AIR at all.

Related tracker link:
https://tracker.adobe.com/#/view/FP-4198417

**Comment by ajwfrost** [comment]
This is an interesting one. "Watson" I think was the name of Adobe's internal bug-tracking system, so they use this to check the status/compatibility levels for bugs. Essentially their aim was to ensure consistency in behaviour, based on the SWF version level that's being run.. so if you built a SWF with version code 15, it would always run the same way, even when the player/runtime was updated and bug fixes applied.

In this particular case - there's only one usage of `watson` from AS3 that I can see - it's a bug related to the Stage3D method `Vector3D.angleBetween`. In Flash Player/AIR versions 15 and earlier, it was possible for there to be a maths error because a calculation ended up outside of the -1 to +1 range, prior to `acos` being called on the value. If you have a SWF that has a way of creating vectors that exhibits this problem, then even if you're running it on our latest runtime, the code would throw a maths error if your SWF version number was 15 or earlier, but it would be okay if you were running a SWF version 16 or later.

The `watson()` method is basically asking the runtime, "should we execute the fixed code for this bug number, or should we execute the original code".

I'm not sure I see the point of this personally.. we do have a few things where behaviour changes for later AIR namespace versions, but for things that are actually bugs, I would think they should always be applied. But we are aware sometimes that changing things - even, fixing things - can cause side effects for people who have carefully worked around the problems!

Hope that helps. And I'm thinking we can delete this one and clean it up... if people really wanted to know if their vector angles function didn't work, they can do the maths for it themselves (and I'm not really sure how it's even possible to end up there..! probably from rounding errors..?!)

thanks
